otter/util/logging_treq.py

```diff
@@ -24,16 +24,23 @@ def _log_request(treq_call, url, **kwargs):
     """
     clock = kwargs.pop('clock', reactor)
     log = kwargs.pop('log', None)
+
     if not log:
         log = default_log
     method = kwargs.get('method', treq_call.__name__)
 
-    treq_transaction = str(uuid4())
+    kwargs.setdefault('headers', {})
+    treq_transaction = kwargs['headers'].setdefault('x-otter-request-id',
+                                                    [str(uuid4())])[0]
```

Comment:
The intention was to actually uses the request ID as the treq transaction ID, if it is provided (if we wanted to overload it.

So possibly:

```python
kwargs.setdefault('headers', {})
treq_transaction = str(uuid.uuid4())
treq_transaction = kwargs['headers'].setdefault('x-otter-request-id', [treq_transaction])[0]
```

Alternately, we can just always overwrite this header, no matter what the user provides (I can't think of a case where we might want to respect this ID after all).